plugin/handler_socket/libhsclient/allocator.hpp

```diff
@@ -42,6 +43,19 @@ typedef std::allocator<int> allocator_type;
 	static_cast<typ *>(malloc((len) * sizeof(typ)))
 #define DENA_ALLOCA_FREE(x) free(x)
 #endif
+*/
+
+// AIX
+#if 1
+#define DENA_ALLOCA_ALLOCATE(typ, len) \
+        static_cast<typ *>((void*)alloca((len) * sizeof(typ)))
+#define DENA_ALLOCA_FREE(x)
+#else
+#define DENA_ALLOCA_ALLOCATE(typ, len) \
+        static_cast<typ *>((void*)malloc((len) * sizeof(typ)))
+#define DENA_ALLOCA_FREE(x) free(x)
+#endif
+
```

Comment:
Why was it needed to make a copy of this block?

Per my understanding alloca() return type is void*, why is it needed to cast void* to void*?